pkg/runner/runner.go

```diff
@@ -153,18 +187,58 @@ func (r *JMeterRunner) Run(execution testkube.Execution) (result testkube.Execut
 	// TODO add additional artifacts to scrape
 	if r.Params.ScrapperEnabled {
 		directories := []string{
-			reportPath,
+			outputDir,
 		}
 
-		err := r.Scraper.Scrape(execution.Id, directories)
-		if err != nil {
+		output.PrintLog(fmt.Sprintf("Scraping directories: %v", directories))
+
+		if err := r.scrape(context.Background(), directories, execution); err != nil {
 			return *executionResult.WithErrors(fmt.Errorf("scrape artifacts error: %w", err)), nil
 		}
 	}
 
 	return executionResult, nil
 }
 
+func (r *JMeterRunner) scrape(ctx context.Context, dirs []string, execution testkube.Execution) (err error) {
+	if !r.Params.ScrapperEnabled {
+		return nil
+	}
+
+	output.PrintLog(fmt.Sprintf("%s Extracting artifacts from %s using filesystem extractor", ui.IconCheckMark, dirs))
+	extractor := scraper.NewFilesystemExtractor(dirs, filesystem.NewOSFileSystem())
+	var loader scraper.Uploader
+	var meta map[string]any
+	if r.Params.CloudMode {
+		output.PrintLog(fmt.Sprintf("%s Uploading artifacts using Cloud uploader", ui.IconCheckMark))
+		meta = cloudscraper.ExtractCloudLoaderMeta(execution)
+		cloudExecutor := cloudexecutor.NewCloudGRPCExecutor(r.CloudClient, r.Params.CloudAPIKey)
+		loader = cloudscraper.NewCloudUploader(cloudExecutor)
+	} else {
+		output.PrintLog(fmt.Sprintf("%s Uploading artifacts using MinIO uploader", ui.IconCheckMark))
+		meta = scraper.ExtractMinIOLoaderMeta(execution)
+		loader, err = scraper.NewMinIOLoader(
+			r.Params.Endpoint,
+			r.Params.AccessKeyID,
+			r.Params.SecretAccessKey,
+			r.Params.Location,
+			r.Params.Token,
+			r.Params.Bucket,
+			r.Params.Ssl,
+		)
+		if err != nil {
+			return err
+		}
+	}
+	scraperV2 := scraper.NewScraperV2(extractor, loader)
```

Comment:
why scrapper V2 - we were discussing it some time ago

Reply:
yeah, old scraper is removed, will drop the V2